sdk/src/sysvars/rent.rs

```diff
@@ -195,22 +176,26 @@ impl Rent {
     ///
     /// The minimum balance in lamports for rent exemption.
     #[inline]
-    #[allow(deprecated)]
     pub fn minimum_balance(&self, data_len: usize) -> u64 {
         let bytes = data_len as u64;
-        let exemption_threshold_as_u64 = u64::from_le_bytes(self.exemption_threshold.to_le_bytes());
+        let exemption_threshold_as_u64 = u64::from_le_bytes(self.exemption_threshold);
 
         match exemption_threshold_as_u64 {
             F64_SIMD0194_EXEMPTION_THRESHOLD_AS_U64 => {
-                (ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte_year
+                (ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte
             }
             F64_EXEMPTION_THRESHOLD_AS_U64 => {
-                ((ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte_year)
+                ((ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte)
                     * DEFAULT_EXEMPTION_THRESHOLD_AS_U64
             }
             _ => {
-                (((ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte_year) as f64
-                    * self.exemption_threshold) as u64
+                #[cfg(any(target_os = "solana", not(target_arch = "bpf")))]
+                {
+                    (((ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte) as f64
+                        * f64::from_le_bytes(self.exemption_threshold)) as u64
+                }
+                #[cfg(target_arch = "bpf")]
+                panic!("Floating-point operations are not supported on BPF targets");
```

Comment:
I am not sure about this, but there is not much we can do if the `exemption_threshold` value does not match one of the known ones.